backend/app/usecase/shortlink/creator.go

```diff
@@ -75,23 +75,24 @@ func (c CreatorPersist) CreateShortLink(createArgs entity.ShortLinkInput, user e
 		return entity.ShortLink{}, ErrMaliciousLongLink(longLink)
 	}
 
-	if !(createArgs.CustomAlias != "") {
+	customAlias := createArgs.GetCustomAlias("")
+	isValid, violation = c.aliasValidator.IsValid(customAlias)
+	if !isValid {
+		return entity.ShortLink{}, ErrInvalidCustomAlias{customAlias, violation}
+	}
+
+	if customAlias == "" {
 		autoAlias, err := c.createAutoAlias()
 		if err != nil {
 			// TODO create error type for fail create auto alias?
 			return entity.ShortLink{}, err
 		}
-		createArgs.CustomAlias = autoAlias
-	} else {
-		isValid, violation = c.aliasValidator.IsValid(createArgs.CustomAlias)
-		if !isValid {
-			return entity.ShortLink{}, ErrInvalidCustomAlias{createArgs.CustomAlias, violation}
-		}
+		customAlias = autoAlias
 	}
 
 	return c.createShortLink(entity.ShortLink{
```

Comment:
Adding this comment to remind myself to update `ShortLink` repository methods to accept `ShortLinkInput` instead of `ShortLink`.